lib/core/pipes/decimal-number.pipe.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DecimalPipe } from '@angular/common';
+import { Pipe, PipeTransform } from '@angular/core';
+import { AppConfigService } from '../app-config/app-config.service';
+import { UserPreferencesService, UserPreferenceValues } from '../services/user-preferences.service';
+import { DecimalNumberModel } from '../models/decimal-number.model';
+
+@Pipe({
+    name: 'adfDecimalNumber',
+    pure: true
+})
+export class DecimalNumberPipe implements PipeTransform {
+
+    static DEFAULT_LOCALE = 'en-US';
+    static DEFAULT_MIN_INTEGER_DIGITS = 1;
+    static DEFAULT_MIN_FRACTION_DIGITS = 0;
+    static DEFAULT_MAX_FRACTION_DIGITS = 2;
+
+    defaultLocale: string = DecimalNumberPipe.DEFAULT_LOCALE;
+    defaultMinIntegerDigits: number = DecimalNumberPipe.DEFAULT_MIN_INTEGER_DIGITS;
+    defaultMinFractionDigits: number = DecimalNumberPipe.DEFAULT_MIN_FRACTION_DIGITS;
+    defaultMaxFractionDigits: number = DecimalNumberPipe.DEFAULT_MAX_FRACTION_DIGITS;
+
+    constructor(public userPreferenceService?: UserPreferencesService,
+                public appConfig?: AppConfigService) {
+
+        if (this.userPreferenceService) {
+            this.userPreferenceService.select(UserPreferenceValues.Locale).subscribe((locale) => {
```

Comment:
should properly unsubscribe to prevent memory leaks, see other use cases on how to use "takeUntil()" pipe

Reply:
True